src/sync/channel.zig

```diff
@@ -153,12 +163,61 @@ pub fn Channel(T: type) type {
                     // Release the exclusive lock on the slot's value, which allows a consumer
                     // to read the data we've just assigned.
                     _ = slot.state.fetchOr(WRITTEN_TO, .release);
+                    channel.condition.signal();
```

Comment:
There is no predictable scheduling ordering between `signal` and `wait`.

> however, if predictable scheduling
       behavior is required, then that mutex shall be locked by the
       thread calling pthread_cond_broadcast() or pthread_cond_signal().

Locking the mutex ensures the other side sees the operation.

Reply:
6966c533dc25412a443b1f9de21ff9e196b57a82